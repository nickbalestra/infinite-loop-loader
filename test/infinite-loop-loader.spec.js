/* eslint-disable no-useless-escape */
'use strict'

const loader = require('../index.js')

describe('infinite-loop-loader', () => {
  it('should be a function with arity 1', () => {
    expect(typeof loader).toBe('function')
    expect(loader.length).toBe(1)
  })

  describe('When invoked on a js file containing loops', () => {
    const source = `
module.exports.data = function(context, cb) {
  var x, y, z;

  while(true) {
    x = 234;
  }

  for(var i = 0; i < 10; i++) {
    y = 456;
  }

  do {
    z = 342;
  } while(false);

  loopWhile:
  while(true) {
    x = 234;
  }

  loopFor:
  for(var i = 0; i < 10; i++) {
    y = 456;
  }

  loopDoWhile:
  do {
    z = 342;
  } while(false);
  return cb(null, data)
};
`
    const result = loader(source)

    it('should wrap WhileStatement, ForStatement and DoWhileStatement correctly', () => {
      expect(result).toMatchSnapshot()
    })
  })
})
