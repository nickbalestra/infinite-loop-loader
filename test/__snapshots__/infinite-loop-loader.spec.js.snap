exports[`infinite-loop-loader When invoked on a js file containing loops should wrap WhileStatement, ForStatement and DoWhileStatement correctly 1`] = `
"
module.exports.data = function(context, cb) {
  var x, y, z;

  
      var __ITER = 1000000000;
      while(true) { if (__ITER <= 0) {
        throw new Error(\"Loop exceeded maximum allowed iterations\");
      }
      __ITER--;
      
    x = 234;
  }

  
      var __ITER = 1000000000;
      for(var i = 0; i < 10; i++) { if (__ITER <= 0) {
        throw new Error(\"Loop exceeded maximum allowed iterations\");
      }
      __ITER--;
      
    y = 456;
  }

  
      var __ITER = 1000000000;
      do { if (__ITER <= 0) {
        throw new Error(\"Loop exceeded maximum allowed iterations\");
      }
      __ITER--;
      
    z = 342;
  } while(false);

  
      var __ITER = 1000000000;
      loopWhile:
  while(true) { if (__ITER <= 0) {
        throw new Error(\"Loop exceeded maximum allowed iterations\");
      }
      __ITER--;
      
    x = 234;
  }

  
      var __ITER = 1000000000;
      loopFor:
  for(var i = 0; i < 10; i++) { if (__ITER <= 0) {
        throw new Error(\"Loop exceeded maximum allowed iterations\");
      }
      __ITER--;
      
    y = 456;
  }

  
      var __ITER = 1000000000;
      loopDoWhile:
  do { if (__ITER <= 0) {
        throw new Error(\"Loop exceeded maximum allowed iterations\");
      }
      __ITER--;
      
    z = 342;
  } while(false);
  return cb(null, data)
};
"
`;
